client/go/outline/config/config_shadowsocks_test.go

```diff
@@ -33,43 +35,131 @@ func parseFromYAMLText(configText string) (*ShadowsocksConfig, error) {
 }
 
 func TestParseShadowsocksConfig_URL(t *testing.T) {
-	t.Run("Fully Encoded", func(t *testing.T) {
-		encoded := base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte("chacha20-ietf-poly1305:SECRET@example.com:1234?prefix=HTTP%2F1.1%20"))
+	t.Run("Fully Base64 Encoded", func(t *testing.T) {
+		encoded := base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte("chacha20-ietf-poly1305:SECRET/!@#@example.com:1234?prefix=HTTP%2F1.1%20"))
 		config, err := parseFromYAMLText("ss://" + string(encoded) + "#outline-123")
 		require.NoError(t, err)
 		require.Equal(t, "example.com:1234", config.Endpoint)
 		require.Equal(t, "chacha20-ietf-poly1305", config.Cipher)
-		require.Equal(t, "SECRET", config.Secret)
+		require.Equal(t, "SECRET/!@#", config.Secret)
 		require.Equal(t, "HTTP/1.1 ", config.Prefix)
 	})
 
-	t.Run("User Info Encoded", func(t *testing.T) {
-		encoded := base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte("chacha20-ietf-poly1305:SECRET"))
+	t.Run("Fully Base64 Encoded With Password Containing Host", func(t *testing.T) {
+		encoded := base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte("chacha20-ietf-poly1305:SECRET@example.com:80@example.com:1234?prefix=HTTP%2F1.1%20"))
+		config, err := parseFromYAMLText("ss://" + string(encoded) + "#outline-123")
+		require.NoError(t, err)
+		require.Equal(t, "example.com:1234", config.Endpoint)
+		require.Equal(t, "chacha20-ietf-poly1305", config.Cipher)
+		require.Equal(t, "SECRET@example.com:80", config.Secret)
+		require.Equal(t, "HTTP/1.1 ", config.Prefix)
+	})
+
+	t.Run("Fully Base64 Encoded Having Query Parameters With Password", func(t *testing.T) {
+		tests := []struct {
+			name    string
+			input   string
+			wantErr bool
+		}{
+			{
+				name:    "Ambiguous Query Parameter",
```

Comment:
None of these are actually ambiguous, because the first `@` is not followed by a hostname, or there's only one `@`. They can all only be parsed one way really.

Please add the ambiguous example I shared as that one will actually parse successfully in two different ways.